Add tuning documentation (#9978)
grafana/mimir: docs/sources/mimir/configure/tuning.md

- Configure [shuffle sharding]({{< relref "./configure-shuffle-sharding" >}})

## Compression

Depending on the CPU model used in the underlying infrastructure, the compression for both WALs and GRPC communication might consumes a significant part of the available CPU resources.
To identify such case one could rely on profiling with tools like [Grafana Pyroscope](https://grafana.com/docs/pyroscope/latest/)

To reduce the resource consumption, users might consider:

- Make sure `wal_compression_enabled` is not enable
- Make sure `grpc_compression` is either off which is the default or configured to `snappy`. `gzip` consumes more CPU than `snappy`. However, disabling `grpc_compression` implies more network traffic and in turn might increase the total cost of ownership (TCO) of running Grafana Mimir.

If users must used compression, like for example to fit in the network bandwidth, they might consider using nodes with more powerful CPU. This implies a increase of the TCO.

## Increase the cache size on a budget

Grafana Mimir relies on Memcached for its caches. Memcached relies, by default only on the memory.

> Similarly to the work of the Ops team behind Grafana Loki, one could enable the `extstore` feature of Memcached.

Reviewer: I don't think it's necessary to include what the Loki team is doing here. Rather, could we provide a brief description as to what enabling `extstore` accomplishes?

> See: [how we scaled Grafana Cloud Logs' Memcached cluster to 50TB and improved reliability](https://grafana.com/blog/2023/08/23/how-we-scaled-grafana-cloud-logs-memcached-cluster-to-50tb-and-improved-reliability/)

Reviewer: ```suggestion
Refer to [how we scaled Grafana Cloud Logs' Memcached cluster to 50TB and improved reliability](https://grafana.com/blog/2023/08/23/how-we-scaled-grafana-cloud-logs-memcached-cluster-to-50tb-and-improved-reliability/).
```

## Periodic latency spikes when cutting blocks

> Depending on the workload, users might witness latency spikes when Grafana Mimir cuts blocks.

Reviewer: ```suggestion
Depending on the workload, you might witness latency spikes when Mimir cuts blocks.
```

> To reduce the impacts of this behavior, users might consider:

Reviewer: ```suggestion
To reduce the impact of this behavior, consider the following:
```

> - Upgrade to `2.15+`, See: <https://github.com/grafana/mimir/commit/03f2f06e1247e997a0246d72f5c2c1fd9bd386df>

Reviewer: ```suggestion
- Upgrade to `2.15+`. Refer to <https://github.com/grafana/mimir/commit/03f2f06e1247e997a0246d72f5c2c1fd9bd386df>.
```

> - Reduce `-blocks-storage.tsdb.block-ranges-period`, default `2h`, For example try `1h`

Reviewer: ```suggestion
- Reduce `-blocks-storage.tsdb.block-ranges-period`, default `2h`. For example. try `1h`.
```
